bin/alluxio-start.sh

```diff
@@ -271,36 +271,44 @@ start_worker() {
       echo "Cache path ${cache} is not a directory; aborting"
       exit 2
     fi
+    if [[ -z $(ls -a "${cache}" ) ]]; then
+      echo "Cache path ${cache} is an empty directory; skipping cache population"
+    else 
+      echo "Populating worker ramcache(s) with contents from ${cache}"
+
+      get_ramdisk_array # see alluxio-common.sh
+      for dir in "${RAMDISKARRAY[@]}"; do
+        if [[ ! -e "${dir}" ]]; then
+          echo "Alluxio has a configured ramcache path ${dir}, but that path does not exist"
+          exit 2
+        fi
+        if [[ ! -d "${dir}" ]]; then
+          echo "Alluxio has a configured ramcache path ${dir}, but that path is not a directory"
+          exit 2
+        fi
 
-    echo "Populating worker ramcache(s) with contents from ${cache}"
+        echo "Populating worker ramcache at ${dir} with ${cache}/${dir}"
+        if [[ ! -e "${cache}/${dir}" ]]; then
+          echo "Path does not exist: ${cache}/${dir}"
+          exit 2
+        fi
+        if [[ ! -d "${cache}/${dir}" ]]; then
+          echo "Path is not a directory: ${cache}/${dir}"
+          exit 2
+        fi
 
-    get_ramdisk_array # see alluxio-common.sh
-    for dir in "${RAMDISKARRAY[@]}"; do
-      if [[ ! -e "${dir}" ]]; then
-        echo "Alluxio has a configured ramcache path ${dir}, but that path does not exist"
-        exit 2
-      fi
-      if [[ ! -d "${dir}" ]]; then
-        echo "Alluxio has a configured ramcache path ${dir}, but that path is not a directory"
-        exit 2
-      fi
+        # recursively delete all (including hidden) files of the ramdisk
+        find "${dir}" -mindepth 1 -delete
 
-      echo "Populating worker ramcache at ${dir} with ${cache}/${dir}"
-      if [[ ! -e "${cache}/${dir}" ]]; then
-        echo "Path does not exist: ${cache}/${dir}"
-        exit 2
-      fi
-      if [[ ! -d "${cache}/${dir}" ]]; then
-        echo "Path is not a directory: ${cache}/${dir}"
-        exit 2
-      fi
-
-      cp -a "${cache}/${dir}/." "${dir}/"
-      if [[ ${?} -ne 0 ]]; then
-        echo "Failed to populate ramcache at ${dir} with ${cache}/${dir}"
-        exit 2
-      fi
-    done
+        # recursively copy all contents of the src directory
+        # (including hidden files) to the destination directory
+        cp -dR "${cache}/${dir}/." "${dir}/"
```

Comment:
Replaced `cp -a` with `cp -dR` since preserving file timestamps was causing 'Operation not permitted' errors. Specifically:
> `cp: preserving times for ‘/mnt/ramdisk/.’: Operation not permitted`

Linux manual page for cp: https://man7.org/linux/man-pages/man1/cp.1.html
- `cp -a` is equivalent to `cp -dR --preserve=all`

Reply:
AFAIK, we don't use timestamps for anything on the worker. So this is safe.

Reply:
I agree about keeping this compliant across all platforms; should I just stick to `-R` then? I doubt the ramdisk directory would have symlinks in it right?

Reply:
I think it's probably safe to drop the `-d` flag. AFAIK we aren't using symlinks in the worker's ramdisk